var/spack/repos/builtin/packages/libedit/package.py

```diff
@@ -30,7 +30,6 @@ class Libedit(AutotoolsPackage):
     homepage = "http://thrysoee.dk/editline/"
     url      = "http://thrysoee.dk/editline/libedit-20150325-3.1.tar.gz"
 
-    version('3.1', '43cdb5df3061d78b5e9d59109871b4f6',
-            url="http://thrysoee.dk/editline/libedit-20150325-3.1.tar.gz")
+    version('20150325-3.1', '43cdb5df3061d78b5e9d59109871b4f6')
```

Comment:
Done. `spack versions` and `spack checksum` are now less useful for this package. Our version detection performance is now down to 98.06%. And I added 4 lines of code which aren't well tested and may break in future releases.